### PR TITLE
Save to perry cog 45

### PR DESCRIPTION
**Pull request by kelfink**
This adds the browser-side calls to submit a PATCH to user details o the rails app.

### CODE REVIEW

File: app/javascript/_services/users/user.service.test.js [hunk 1]
```diff
@@ -20,11 +20,39 @@ describe('UserService', () => {
...
+    });
 
     it('calls #fetchUsersDetails ApiService', () => {
       getSpy.mockReturnValue(Promise.resolve({}));
       UserService.fetchUserDetails(id);
       expect(getSpy).toHaveBeenCalledWith('/user_detail/id');
     });
   });
+
+  describe('#savehUserDetails', () => {
```
Comment (Sushmareddy-OSI) [inline app/javascript/_services/users/user.service.test.js]: spellcheck "#saveUserDetails"

Comment (RamuKammagani-osi) [reply app/javascript/_services/users/user.service.test.js]: +1

File: app/javascript/_services/users/user.service.test.js [hunk 2]
```diff
@@ -20,11 +20,41 @@ describe('UserService', () => {
...
+    let patchSpy;
+
+    beforeEach(() => {
+      patchSpy = jest.spyOn(ApiService, 'patch');
+    });
+
+    it('calls #fetchUsersDetails ApiService', () => {
```
Comment (RamuKammagani-osi) [inline app/javascript/_services/users/user.service.test.js]: #saveUserDetails

File: app/javascript/views/userDetail/UserDetail.jsx [hunk 3]
```diff
@@ -37,6 +39,13 @@ export default class UserDetail extends Component {
     });
   };
 
+  onSaveDetails = () => {
+    const id = getUserId(currentPathname());
+    const { details } = this.state;
+    const response = UserService.saveUserDetails(id, details);
```
Comment (RamuKammagani-osi) [inline app/javascript/views/userDetail/UserDetail.jsx]: Why UserService is called in UserDetail Component, Shouldn't be this in Saga?? I think onClick function should fire up an action and based on the action, patch request should happen! i guess that's gonna happen in Saga's. Isn't it??

Comment (kelfink) [reply app/javascript/views/userDetail/UserDetail.jsx]: Due to the complexity I punted on saga last sprint in favor of just  getting it to go directly.  How do you feel about me adding a tech debt item to transform it to saga ?

Comment (RamuKammagani-osi) [reply app/javascript/views/userDetail/UserDetail.jsx]: I'm okay with that!

**Comment by RamuKammagani-osi** [comment]
retest this please

**Pull request merged**